### ISSUE TITLE
Cmd java error "Segmentation fsult"?

### ISSUE THREAD

**Issue by Biangkerok32** [MasterDevX/Termux-Java#2]
My java us error?
Im use arm device,install succesful..
But java cmd not work?
![Screenshot_2019-06-05-08-20-46](https://user-images.githubusercontent.com/48493593/58923696-e9c81d80-876a-11e9-9163-3194616ba23a.png)

**Comment by MasterDevX** [comment]
Maybe this can help you:
https://github.com/termux/termux-packages/issues/2129

**Comment by Biangkerok32** [comment]
im cant fix this,,,
this reference https://github.com/termux/termux-packages/issues/2129 is diference (i thinks) ,,,
any idea again ,,,?

**Comment by MasterDevX** [comment]
Not sure, gonna make some research tomorrow.

**Comment by OmrAmer** [comment]
MasterDevX , Can you tell us how to install it manually?

**Comment by MasterDevX** [comment]
I think I'll write it in ReadMe soon, but this issue is not an installation related problem, it's execution problem, so..

**Comment by OmrAmer** [comment]
Did you find a solution

**Comment by MasterDevX** [comment]
Not yet.

**Comment by ZalgoNoise** [comment]
Bump. 

I have the understanding that, with this same issue, the commands run when called as root (with tsudo prefixed). 

I have noticed the Compiler (javac) works - it created Main.class. I might be too noob with Java, but executing a Java file with one method with System.out.println() didn't succeed (cant find the designated Class, regardless how you call it - $ tsudo java Main.java ; $ tsudo java Main ;  $ tsudo java Main.class)

**Comment by JanisPlayer** [comment]
Is today known where the error comes from?

**Comment by TudbuT** [comment]
Same

**Comment by JanisPlayer** [comment]
> Same

I've found that it needs root privileges for this Java.
userLand or Ubuntu for XTerm(Need Tunnel) can be an alternative.  If you're rootest I can deploy Linux to you.

**Comment by MasterDevX** [comment]
Can you try to install proot and execute proot -0 before running Java?

**Comment by TudbuT** [comment]
> Can you try to install `proot` and execute `proot -0` before running Java?

Nope didn't work (also the termux/termux-packages#2129 doesn't help at all)...

**Comment by TudbuT** [comment]
YES!! I spent an hour searching for methodes but found no working one and
proot  -0 only starts the system terminal and I DON'T WANT TO ROOT MY
DEVICE! So yes I tried that boy...

MasterDevX <notifications@github.com> schrieb am Do., 7. Nov. 2019 07:06:

**Comment by MasterDevX** [comment]
@TudbuT PRoot is not rooting your device..

**Comment by KingRanger984** [comment]
It is an issue with Android Version! I have seen people using Hax4us' OpenJDK 8 on Android Version 5, I guess, and it worked! On other Android Versions, segmentation fault is common!

Only One Solution is Available - Install Ubuntu(Or Any Other Distribution) Chroot (of course it's PRoot in Termux) and use OpenJDK there.

**Comment by KingRanger984** [comment]
> Bump.
> 
> I have the understanding that, with this same issue, the commands run when called as root (with tsudo prefixed).
> 
> I have noticed the Compiler (javac) works - it created Main.class. I might be too noob with Java, but executing a Java file with one method with System.out.println() didn't succeed (cant find the designated Class, regardless how you call it - $ tsudo java Main.java ; $ tsudo java Main ; $ tsudo java Main.class)
> 
> ![Screenshot_20190809-015457_Termux](https://user-images.githubusercontent.com/23582419/62746677-cc545000-ba48-11e9-8281-78dfcef18467.jpg)

Rooting won't help! It is a problem with Android version.

**Comment by commander-cb** [comment]
i had the same problem
the script throws a seg fault
so i installed these instead


openjdk-9-jdk-headless_9.2017.8.20-1_arm.deb
openjdk-9-jre-headless_9.2017.8.20-1_arm.deb


must be on line
apt-get install -y ./openjdk-9-jdk-headless_9.2017.8.20-1_arm.deb
and the other one .  i will let you figure out the rest okay

**Comment by Jaytixx** [comment]
How exactly do I uninstall it please your directives were not clear on how to do that?
